File: 171-excel-sheet-column-number/171-excel-sheet-column-number.js
/**
 * @param {string} columnTitle
 * @return {number}
 */
const titleToNumber = function(c) {
    let result = 0;
    const ACode = 'A'.charCodeAt(0);
    for (let i = 0; i < c.length; i++) {
        const charVal = c[c.length - 1 - i].charCodeAt(0) - ACode + 1; 
        result += charVal * (26**i);
    }
    return result;
};

File: 171-excel-sheet-column-number/171-excel-sheet-column-number.rb
# @param {String} column_title
# @return {Integer}
def title_to_number(column_title)
    offset = 'A'.ord - 1
    res = 0
    0.upto(column_title.size - 1) do |i|
        char = column_title[-1 - i]
        char_val = char.ord - offset
        res += 26**i * char_val
    end
    res
end